File: apps/dot/frontend/src/lib/api/usage.ts
import { apiClient } from '@/lib/api/api-client'
import type { GetAccessToken } from '@/lib/api/api-client'

export type UsagePeriod = {
  start: string
  end: string
}

export type ProviderModelCost = {
  [modelId: string]: number
}

export type ProviderBreakdownItem = {
  provider: string
  total_usd: number
  tokens_in: number
  tokens_out: number
  models: ProviderModelCost
}

export type UsageBreakdown = {
  chat_usd: number
  reasoning_usd: number
  vision_usd: number
  image_usd: number
}

export type UsageDailyItem = {
  date: string
  usd: number
}

export type UsageSummary = {
  cliente_id: string
  period: UsagePeriod
  limit_usd: number
  consumed_usd: number
  consumed_percent: number
  remaining_usd: number
  limit_enabled: boolean
  blocked: boolean
  breakdown?: UsageBreakdown
  provider_breakdown?: ProviderBreakdownItem[]
  projected_depletion_date?: string | null
}

export type UsageDailyResponse = {
  days: UsageDailyItem[]
}

export async function fetchUsageSummary(
  getAccessToken: GetAccessToken,
): Promise<UsageSummary> {
  return apiClient.get<UsageSummary>('/v1/usage/summary', getAccessToken)
}

export async function fetchUsageDaily(
  getAccessToken: GetAccessToken,
  days = 7,
): Promise<UsageDailyResponse> {
  return apiClient.get<UsageDailyResponse>(`/v1/usage/daily?days=${days}`, getAccessToken)
}

File: apps/dot/backend/app/schemas/usage.py
"""Esquemas de respuesta para consumo IA."""
from __future__ import annotations

from pydantic import BaseModel, Field


class UsagePeriodResponse(BaseModel):
    start: str = Field(description="Inicio del mes de facturación (YYYY-MM-DD)")
    end: str = Field(description="Fin del mes de facturación (YYYY-MM-DD)")


class UsageBreakdownResponse(BaseModel):
    chat_usd: float = 0.0
    reasoning_usd: float = 0.0
    vision_usd: float = 0.0
    image_usd: float = 0.0


class UsageDailyItemResponse(BaseModel):
    date: str = Field(description="Fecha ISO (YYYY-MM-DD)")
    usd: float = Field(description="Costo en USD de ese día")


class UsageDailyResponse(BaseModel):
    days: list[UsageDailyItemResponse]


class UsageSummaryResponse(BaseModel):
    cliente_id: str
    period: UsagePeriodResponse
    limit_usd: float
    consumed_usd: float
    consumed_percent: int
    remaining_usd: float
    limit_enabled: bool
    blocked: bool
    breakdown: UsageBreakdownResponse | None = None
    projected_depletion_date: str | None = Field(
        None,
        description="Fecha estimada de agotamiento (DD/MM) o null si no se agota este mes",
    )
